Description: Add-on to cups-avahi.patch to support Apple AirPrint (printing from iPhone, iPad, iPod Touch to a CUPS server), patch is separate as this part is not in the upstream repository of cups-avahi.patch because it is not intencded to be submitted to CUPS upstream.
Author: Till Kamppeter <till.kamppeter@gmail.com>
Bug: https://bugs.launchpad.net/bugs/711779

Index: cups-1.6.0/scheduler/conf.c
===================================================================
--- cups-1.6.0.orig/scheduler/conf.c	2012-07-27 16:55:12.000000000 +0200
+++ cups-1.6.0/scheduler/conf.c	2012-07-27 17:15:59.550185245 +0200
@@ -738,7 +738,7 @@
   DefaultShared            = CUPS_DEFAULT_DEFAULT_SHARED;
 
 #ifdef HAVE_DNSSD
-  cupsdSetString(&DNSSDSubTypes, "_cups,_print");
+  cupsdSetString(&DNSSDSubTypes, "_cups,_print,_universal");
 #endif /* HAVE_DNSSD */
 
   cupsdSetString(&LPDConfigFile, CUPS_DEFAULT_LPD_CONFIG_FILE);
Index: cups-1.6.0/scheduler/dirsvc.c
===================================================================
--- cups-1.6.0.orig/scheduler/dirsvc.c	2012-05-18 04:25:18.000000000 +0200
+++ cups-1.6.0/scheduler/dirsvc.c	2012-07-27 17:18:49.966182963 +0200
@@ -606,6 +606,9 @@
     keyvalue[count  ][0] = "pdl";
     keyvalue[count++][1] = p->pdl ? p->pdl : "application/postscript";
 
+    keyvalue[count  ][0] = "URF";
+    keyvalue[count++][1] = "none";
+
     if (get_auth_info_required(p, air_str, sizeof(air_str)))
     {
       keyvalue[count  ][0] = "air";
